src/features/unified-share-modal/SharedLinkSection.js

```diff
@@ -130,6 +132,8 @@ class SharedLinkSection extends React.Component<Props, State> {
 
         if (!prevProps.sharedLink.url && sharedLink.url) {
             this.setState({ isAutoCreatingSharedLink: false });
+        } else if (!sharedLink.url && this.toggleRef) {
```

Comment:
This logic will still focus on the toggle when the modal first loads. Since you want to focus on the toggle only when going from deactivated (!prevProps.sharedLink.url) to activated (sharedLink.url), you want to nest it under the condition above:
```
if (!prevProps.sharedLink.url && sharedLink.url) {
    this.setState({ isAutoCreatingSharedLink: false });
    if (this.toggleRef) {
        this.toggleRef.focus();
    }
}
```

Reply:
@swfree I consider the logic is OK because I want to focus on the toggle only when `sharedLink.url` is null or empty it means when I click in shared button and it does not auto generate a link. You're right when you say that the SharedLinkSection first load it will focus to the toggle, however it will focus to `TextInputWithCopyButton` when  `sharedLink.url` changes from null or empty to any other value and it will work as expected. 

I did this change for recommendation of WebAIM in order to focus the most important feature in the modal which is Share Link toggle only when `sharedLink.url` does not exist instead to focus on invite people input.